### ISSUE TITLE
Replace file() with open() in twisted/internet

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8527]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8527|
|Type|enhancement|
|Created|2016-06-28 06:18:19Z|
|Branch|https://github.com/twisted/twisted/tree/8527-rodrigc-file|

In Python 3, file() is gone (see https://docs.python.org/3/whatsnew/3.0.html#builtins ), and needs to be replaced with open()
which works on Python 2 and 3

<details><summary>Searchable metadata</summary>

```
trac-id__8527 8527
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8527_rodrigc_file 8527-rodrigc-file
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__None None
time__1467094699177003 1467094699177003
changetime__1468062227168355 1468062227168355
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/253

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_2"></a>|@Lukasa set owner to @rodrigc|
|-|-|

Looks good to me, go ahead and merge when ready.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc commented|
|-|-|

Lukasa, I can't merge this one because codecov says there is no coverage.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub set status to `closed`|
|-|-|

In changeset 177052b0ff1eb140b03c728a1f1caacb431fa998
```
#!CommitTicketReference repository="" revision="177052b0ff1eb140b03c728a1f1caacb431fa998"
Merge pull request #253 from twisted/8527-rodrigc-file

Author: rodrigc
Reviewer: Lukasa, alex
Fixes: #8527

Replace file() with open() in twisted/internet
```
